### DISCUSSION TITLE
CI: removing the custom ci Task helper

### DISCUSSION THREAD

**Discussion by mockersf** [bevyengine/bevy#3062] [General]
By using a custom launcher (`cargo run -p ci`) for `cargo fmt` and `cargo clippy`, we're missing on better integrations with GitHub. For example, the clippy action https://github.com/actions-rs/clippy-check annotates issues directly in the commit (see their screenshot).

Personally, I never use the `cargo run -p ci` command, and I find it harder to read in the GitHub action logs. I think it doesn't add that much value as we're just running format and clippy, and no custom actions.

For reference, it was added in #1387.

I want to remove it, but would like others point of view before opening that PR.

Please vote, 👍 for "ok to remove", 👎 for "no I want to keep it"

**Comment by TheRawMeatball** [comment]
I'd be strongly opposed to removing it, as it makes it drastically easier to validate CI locally (which I do every now and then).

**Reply by mockersf** [reply]
`cargo run -p ci` has extra checks now which makes it more useful

**Comment by shanesveller** [comment]
Would [cargo-xtask](https://github.com/matklad/cargo-xtask) split the middle and offer local extensibility and structure while possibly remaining compatible with more surrounding components? Or does it share most of the same concerns?

**Reply by mockersf** [reply]
Bevy ci package uses the same idea than cargo-task, but without the alias

**Comment by bjorn3** [comment]
> For example, the clippy action https://github.com/actions-rs/clippy-check annotates issues directly in the commit (see their screenshot).

I believe the problem matcher used by clippy-check is applied to all further build steps even if they don't use clippy-check. Adding a dummy actions-rs/check step at the start (eg building just the ci package, as that one should be quick to build) may be enough to get annotations for `cargo run -p ci`.

**Reply by mockersf** [reply]
clippy-check also uses a token and some code to add the annotations at the correct places in the files view of a pr
